src/permutat.cc

```diff
@@ -67,6 +69,36 @@
 #include "sysfiles.h"
 #include "trans.h"
 
+}
+
+template<typename T> static inline UInt SIZEBAG_PERM(UInt deg)
+{
+    return sizeof(Obj) + deg * sizeof(T);
+}
+
```

Comment:
Might be nice to have some kind of `CHECK_PERM_TYPE<T>(Obj)` function, that could be used with GAP_ASSERT to check if `T` is the right kind of type to access `Obj` Something like (sorry, not checked with compiler)

```
template<typename T> static inline CHECK_PERM_TYPE<T>(Obj) { return false; }
template<> static inline CHECK_PERM_TYPE<UInt2>(Obj o) { return TNUM_OBJ(o) == T_PERM2; }
template<> static inline CHECK_PERM_TYPE<UInt4>(Obj o) { return TNUM_OBJ(o) == T_PERM4; }
```